consensus/common_test.go

```diff
@@ -37,6 +37,23 @@ func ensureDir(dir string, mode os.FileMode) {
 	}
 }
 
+type NodeConfig struct {
+	cfg.Config `mapstructure:",squash"`
+	P2P        *p2p.Config   `mapstructure:"p2p"`
+	Mempool    *mempl.Config `mapstructure:"mempool"`
+	Consensus  *Config       `mapstructure:"consensus"`
+}
+
+// TODO: This is the same as NodeConfig. Should we move the configs to types (?)
```

Comment:
Yes I think that's reasonable and was actually my initial intuition. But I was lured by the appeal of having config structs defined where they are used (something about being able to look for the consensus config in the consensus package just feels right). As right as that feels, it makes other things feel quite wrong, so probably best to put all the config together in types/config.go. So we'll have `Config`, `ConsensusConfig`, `MempoolConfig`, `P2PConfig`, and then `NodeConfig` will include all the above ?

I should have put this all down in an ADR. I'm doing that now too